lib/browser/api/power-monitor.js

```diff
@@ -5,4 +5,16 @@ const {powerMonitor, PowerMonitor} = process.atomBinding('power_monitor')
 Object.setPrototypeOf(PowerMonitor.prototype, EventEmitter.prototype)
 EventEmitter.call(powerMonitor)
 
+powerMonitor.on('newListener', (event) => {
+  if (event === 'shutdown' && powerMonitor.listenerCount('shutdown') === 0) {
+    powerMonitor.blockShutdown()
+  }
+})
+
+powerMonitor.on('removeListener', (event) => {
+  if (event === 'shutdown' && powerMonitor.listenerCount('shutdown') === 0) {
```

Comment:
Is a "shutdown" listener removed _before_ this function is called?

Reply:
Yes